app/lib/features/notifications/providers/notifiers/notification_settings_notifier.dart

```diff
@@ -10,21 +10,26 @@ final _log = Logger('a3::common::notification_settings_notifier');
 
 class AsyncNotificationSettingsNotifier
     extends AsyncNotifier<NotificationSettings> {
-  late Stream<bool> _listener;
-  late StreamSubscription<bool> _poller;
+  // ignore: unused_field
+  Stream<bool>? _listener;
+  StreamSubscription<bool>? _poller;
 
   @override
   Future<NotificationSettings> build() async {
+    ref.onDispose(() => _poller?.cancel());
```

Comment:
this PR revealed a bug in this notifier which lead to it only updating once. The reason was that we need to re-subscribe after each time when we create a new settings object as the old one gets dropped. the new code does that: triggering reset will always subscribe to the new one and add a listener to call reset again.